FirebaseSwiftUI/FirebaseAuthSwiftUI/Sources/FirebaseAuthSwiftUI/FUIAuth.swift

```diff
@@ -0,0 +1,197 @@
+  }
+
+  public func authProviders(providers: [FUIAuthProvider]) {
+    authProviders = providers
+  }
+
+  public func getEmailProvider() -> EmailAuthProvider? {
+    return try! providerWithId(providerId: "email") as! EmailAuthProvider
+  }
+
+  public func providerWithId(providerId: String) -> FUIAuthProvider? {
+    if let provider = authProviders.first(where: { $0.providerId == providerId }) {
+      return provider
+    } else {
+      assertionFailure(
+        "Provider with ID \(providerId) not found. Did you add it to the authProviders array?"
+      )
+      return nil
+    }
+  }
+}
+
+// main auth view - can be composed of custom views or fallback to default views. We can also pass
+// state upwards as opposed to having callbacks.
+// Negates the need for a delegate used in UIKit
+public struct FUIAuthView<Modifier: ViewModifier>: View {
+  private var authFUI: FUIAuth
+  private var authPickerView: AuthPickerView<Modifier>
+
+  public init(FUIAuth: FUIAuth,
+              _authPickerView: AuthPickerView<Modifier>? = nil) {
+    authFUI = FUIAuth
+    authPickerView = _authPickerView ?? AuthPickerView()
+  }
+
+  public var body: some View {
+    VStack {
+      AnyView(authPickerView)
+    }
+  }
+}
+
+public protocol AuthPickerViewProtocol: View {
+  var title: AnyView { get }
+}
+
+public struct AuthPickerModifier: ViewModifier {
+  public func body(content: Content) -> some View {
+    content
+      .padding(20)
+      .background(Color.white)
+      .cornerRadius(12)
+      .shadow(radius: 10)
+      .padding()
+  }
+}
+
+public struct AuthPickerView<Modifier: ViewModifier>: AuthPickerViewProtocol {
+  private var emailAuthButton: EmailAuthButton
+  private var vStackModifier: Modifier
+
+  public init(title _: String? = nil, _emailAuthButton: EmailAuthButton? = nil,
+              _modifier: Modifier? = nil) {
+    emailAuthButton = _emailAuthButton ?? EmailAuthButton()
+    vStackModifier = _modifier ?? AuthPickerModifier() as! Modifier
+  }
+
+  public var body: some View {
+    VStack {
+      title
+      emailAuthButton
+    }.modifier(vStackModifier)
+  }
+
+  // Default implementation that can be overridden
+  public var title: AnyView {
+    AnyView(
+      Text("Auth Picker view")
+        .font(.largeTitle)
+        .padding()
+    )
+  }
+}
+
+public protocol FUIButtonProtocol: View {
+  var buttonContent: AnyView { get }
+}
+
+public struct EmailAuthButton: FUIButtonProtocol {
+  @State private var emailAuthView = false
+  public var body: some View {
+    VStack {
+      // TODO: - update FUIButtonProtocol with ways to align the button/have defaults
+      Button(action: {
+        emailAuthView = true
+      }) {
+        buttonContent
+      }
+      NavigationLink(destination: EmailEntryView(), isActive: $emailAuthView) {
+        EmptyView()
+      }
+    }
+  }
+
+  // Default implementation that can be overridden
+  public var buttonContent: AnyView {
+    AnyView(
+      Text("Sign in with email")
```

Comment:
I recommend using `Button` and (if required) define custom button styles. See https://peterfriese.dev/blog/2023/swiftui-styling-views/